ng-dev/pr/merge/pull-request.ts

```diff
@@ -106,16 +112,34 @@ export async function loadAndValidatePullRequest(
     !!config.pullRequest.caretakerNoteLabel &&
     labels.includes(config.pullRequest.caretakerNoteLabel);
 
+  /** Number of commits in the pull request. */
+  const commitCount = prData.commits.totalCount;
+  /**
+   * SHA for the first commit the pull request is based on.
+   *
+   * Typically we would be able to rely on referencing the the base revision as the temprorary
+   * pull request head commmit minus the number of commits in the pull request. This is not
+   * always the case when we rebase the PR with autosquash where the amount of commits could
+   * change. We avoid this issue around this by parsing the base revision so that we are able
+   * to reference a specific SHA before a autosquash rebase could be performed.
+   */
+  const baseSha = git.run(['rev-parse', `${TEMP_PR_HEAD_BRANCH}~${commitCount}`]).stdout.trim();
+  /* Git revision range that matches the pull request commits. */
+  const revisionRange = `${baseSha}..${TEMP_PR_HEAD_BRANCH}`;
+
   return {
     url: prData.url,
     prNumber,
     labels,
     requiredBaseSha,
     githubTargetBranch,
     needsCommitMessageFixup,
+    commitCount,
+    baseSha,
+    revisionRange,
     hasCaretakerNote,
     targetBranches: target.branches,
+    cherryPickTargetBranches: target.branches.filter((b) => b !== githubTargetBranch),
```

Comment:
Since this is now only used in the API merge, we can keep defining this in the API strategy there. Mostly to avoid duplicating the target branches in two places as part of the `PullRequest` interface